src/library/scala/collection/mutable/HashSet.scala

```diff
@@ -25,7 +25,6 @@ final class HashSet[A]
 
   // Used by scala-java8-compat (private[mutable] erases to public, so Java code can access it)
   private[mutable] def getTable: FlatHashTable[A] = table
-
   override def iterator: Iterator[A] = table.iterator
```

Comment:
This seems to unconditionally allocate iterator also for empty: https://github.com/retronym/scala/blob/ec5d494534415fe8d4f2ecdec3bdb22c3a0d9487/src/library/scala/collection/mutable/FlatHashTable.scala#L197